apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java

```diff
@@ -123,14 +128,16 @@ private static JsonSchema getSchema(String resource) {
     }
 
     /**
-     * Sets all optional checks to their default value (enabled), should be used as a shortcut to reset mock reporter state
-     * after/before using it for a single test execution
+     * Resets checks to be executed based on the default behaviour.
+     * All optional checks are enabled and special requested behaviour is disabled.
```

Comment:
[minor] we could be more specific here by replacing 'special requested behavior' with `GC execution before recycling checks`.

Reply:
It's written generically on purpose, as we keep adding nuances to our tests that are being reflected here. Similarly, we don't elaborate what the optional checks are.